03_building_and_packaging/pip_demo.md

```diff
@@ -0,0 +1,87 @@
+# `pip`: Python Packaging Tool (Notes)
+
+**Note**: Maybe run everything in a fresh Docker container. I built the container `jaustar/spack-package-tutorial-base` locally for myself. It contains installed/downloaded Spack, that was not set up yet, i.e., one needs to run `spack compiler find` and `spack external find`. `clingo` needs to be downloaded first as well.
```

Comment:
yes, i would recommend to use a completely fresh Docker container or a Python `venv`.

Reply:
I am still hesitant to use Docker here, because the students have not learnt Docker yet and I do not want to start using things which have not been taught. I will suggest `venv` here